styled/modal.js

```diff
@@ -1,19 +1,43 @@
 import styled from 'styled-components/native';
 import Colors from '../assets/Colors';
+import { Subhead } from '../components/BaseComponents';
 import { ColumnContainer, RowContainer } from './shared';
 
 // Modals in CheckoutScreen.js
 
+// top: -200px is temp fix for avoiding the keyboard
 export const ModalContentContainer = styled(ColumnContainer)`
-  height: 40%;
-  width: 45%;
+  width: ${props => props.width || '45%'};
+  height: ${props => props.height || '40%'};
   margin: auto;
   justify-content: space-around;
   align-items: center;
   background-color: white;
-  top: -200px;
 `;
 
+export const ModalCenteredOpacityLayer = styled(RowContainer)`
+  height: 100%;
+  width: 100%;
+  background-color: rgba(0, 0, 0, 0.3);
+  justify-content: center;
+  align-items: center;
+`;
+
+export const ModalCopyContainer = styled(ColumnContainer)`
+  justify-content: space-around;
+  align-items: ${props => props.alignItems || 'flex-start'};
+`;
+
+// Rewards
+export const SubheadSecondary = styled(Subhead)`
+  color: ${Colors.secondaryText};
+`;
+
+export const SubheadActive = styled(Subhead)`
+  color: ${Colors.activeText};
+`;
```

Comment:
Why not just directly use the base component?

Reply:
It was really ugly in the file since it was used so often LOL